src/pkgcheck/checks/eclass.py

```diff
@@ -254,41 +277,97 @@ def check_provided_eclasses(self, pkg, inherits: list[tuple[list[str], int]]):
         for provided, (eclass, lineno) in provided_eclasses.items():
             yield ProvidedEclassInherit(eclass, pkg=pkg, line=provided, lineno=lineno)
 
+    def check_exported_eclass_phase(
+        self, pkg: bash.ParseTree, inherits: list[tuple[list[str], int]]
+    ):
+        """Check for eclasses exporting the same phase where the ebuild does not
+        call such phases manually."""
+        latest_eapi = EAPI.known_eapis[sorted(EAPI.known_eapis)[-1]]
+        # all known build phases, e.g. src_configure
+        known_phases = list(latest_eapi.phases_rev)
+        exported_phases = {phase: [] for phase in known_phases}
+
+        # Create a dict of known phases => eclasses exporting them:
+        # we're interested in the cases where the RHS list has > 1 element.
+        for eclasses, _ in inherits:
+            for eclass in eclasses:
+                for func in self.eclass_cache[eclass].functions:
+                    phase = func.name.removeprefix(f"{eclass}_")
+                    if phase in known_phases:
+                        exported_phases[phase].append(eclass)
+
+        if not exported_phases.keys():
+            return
+
+        defined_phases = []
+        for node in bash.func_query.captures(pkg.tree.root_node).get("func", ()):
+            func_name = pkg.node_str(node.child_by_field_name("name"))
+            if func_name in known_phases:
+                defined_phases.append(func_name)
+
+        # XXX: Some eclasses vary their API based on the EAPI, usually to
+        # 'unexport' a phase. self.eclass_cache is generated once per eclass,
+        # not (eclass, EAPI), so it can't handle this. Ditto phases which are only
+        # exported if a variable is set. Blacklist such eclasses here as it
+        # doesn't happen often.
+        #
+        # We could maybe make this more finely-grained for phases we know
+        # are conditionally exported if this list is impacting coverage
+        # severely.
+        blacklisted_eclasses = ["pypi", "vala", "xdg"]
```

Comment:
Suppressions like this need to be a class member so testing code can vary it.  Plus it's easier to spot. :)

That said- I assume we can rip out a simple literal asignment out of the parsed bash?  I've not played w/ tree-sitter (yet), but I'm hoping we can reliable scrape a global assignment like this, out of  an eclass:
```
_PKGCHECK_DISABLED_QA_CHECKS_=( ShadowedEclassPhase OtherCheck1 OtherCheck2 )
```
I'm suggesting an array specifically so other checks can move their suppression lists into the source itself.  

Doing this also allows external eclasses's to get in on the game- I'm assuming somewhere buried in the check pathways there is a `if it's not the gentoo repo, don't run me` ?